sgkit/window.py

```diff
@@ -43,12 +49,84 @@ def window(
     -------
     A dataset containing the following variables:
 
+    - :data:`sgkit.variables.window_contig_spec` (windows):
+      The index values of window contigs.
     - :data:`sgkit.variables.window_start_spec` (windows):
       The index values of window start positions.
     - :data:`sgkit.variables.window_stop_spec` (windows):
       The index values of window stop positions.
     """
     step = step or size
+    return _window_per_contig(ds, variant_contig, merge, _get_windows, size, step)
+
+
+window = window_by_index
+
+
+def window_by_position(
+    ds: Dataset,
+    *,
+    size: int,
+    offset: int = 0,
```

Comment:
could we add a ``num`` (or similar) option which is mutually exclusive to ``size`` that gives the number of windows? This would be really useful for plotting - I want to plot around 200 windows, e.g. 

So the ``size`` argument corresponds to numpy.arange and the ``num`` argument corresponds to ``numpy.linspace``? 

Haven't thought through how ``offset`` fits in here.

Reply:
`window_by_index` uses `np.arange`, so perhaps a `num` argument makes more sense there?

Reply:
Yes, sorry, this is what I meant

Reply:
You could want "num" in either - maybe I want ``n`` windows with the same number of variants (or as close as possible) or I want ``n`` equally spaced windows along the genome. Both are valid things, I think, and would be useful.

Reply:
True. I think we could add `num` later, since it's fairly easy to work around. For `window_by_index`, you can set `size=ds.dims["variants"] // num`. For `window_by_position` you can work out the size by adding up the lengths of the contigs and dividing by `num`.

However, this does raise a difference in the types of windowing when measuring by genomic position. The `window_by_position` function here produces one window per variant (position), which is what we want for computing `ld_matrix` (which came from Eric's implementation, and was influenced by Hail's [locus_windows](https://hail.is/docs/0.2/linalg/utils/index.html#hail.linalg.utils.locus_windows)). This is subtly different to chopping the genome up into equally spaced windows measured by number of base pairs. I think we need both, but I'm not sure what the second one looks like from an API point of view (or what it should be called!) - particularly as it probably involves #434 as well.

Reply:
Thinking about this more, I think the two ways of looking at windowing when measuring by genomic position can be generalized. I've introduced a variable called `window_start_position` that can be used to set window starts - so setting it to `variant_position` for example will create a window per variant (the `ld_matrix` case). Alternatively, leaving it unset will produce equally-spaced windows of size `size` across the genome (Jerome's case).

How does this look?